packages/shared/src/hooks/useChat.ts

```diff
@@ -130,26 +133,28 @@ export const useChat = ({ id: idFromProps }: UseChatProps): UseChat => {
   );
 
   const executePrompt = useCallback(
-    (value: string) => {
+    async (value: string) => {
       if (!value) {
         return;
       }
 
-      if (endStreamRef.current) {
-        endStreamRef.current();
+      if (sourceRef.current?.OPEN) {
+        sourceRef.current.close();
       }
 
       setSearchQuery(undefined);
-      // eslint-disable-next-line @typescript-eslint/no-use-before-define
-      endStreamRef.current = mockChatStream({ onMessage });
+      const source = await sendSearchQuery(value, accessToken.token);
+      source.onmessage = onMessage;
+      source.onerror = () => source.close();
```

Comment:
Just keep in mind we need to show a error as well.
You can log it seperate if you want.

Reply:
Yep, we should have the [ticket](https://dailydotdev.atlassian.net/jira/software/projects/WT/boards/11?selectedIssue=WT-1678) dedicated for it.